src/Microsoft.AspNetCore.Mvc.Core/Internal/ContentResultExecutor.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    public class ContentResultExecutor
+    {
+        /// <summary>
+        /// The maximum number of characters that are encoded and written to stream in a single pass.
+        /// </summary>
+        public const int MaxCharacterChunkSize = 1024;
+
+        private readonly string DefaultContentType = new MediaTypeHeaderValue("text/plain")
+        {
+            Encoding = Encoding.UTF8
+        }.ToString();
+
+        private readonly ILogger<ContentResultExecutor> _logger;
+
+        public ContentResultExecutor(ILogger<ContentResultExecutor> logger)
```

Comment:
Sure, I noticed that we already register the following in DI..will fix
https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/MvcCoreServiceCollectionExtensions.cs#L207-L208